sklearn/ensemble/_hist_gradient_boosting/tests/test_binning.py

```diff
@@ -21,8 +22,12 @@ def _find_binning_thresholds(data, max_bins=255, subsample=int(2e5),
                              random_state=None):
     # Just a redef to avoid having to pass arguments all the time (as the
     # function is private we don't use default values for parameters)
-    return _find_binning_thresholds_orig(data, max_bins, subsample,
-                                         random_state)
+    rng = check_random_state(random_state)
+    subsample = subsample
+    if subsample is not None and data.shape[0] > subsample:
+        subset = rng.choice(data.shape[0], subsample, replace=False)
+        data = data.take(subset, axis=0)
+    return _find_binning_thresholds_orig(data, max_bins)
```

Comment:
Tests will pass withouth subsampling now. Sounds like we can remove this helper.

If that involves too many unrelated changes, maybe mark it as a todo